byway/src/main/webapp/routepage.js

```diff
@@ -12,12 +12,26 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// copies of recommendations and selected stops, stored as arrays
+// for synchronous updating
+let recs = [];
+let stops = [];
+
+let l = load;
+
 if (document.readyState === 'loading') {  // Loading hasn't finished yet
-  document.addEventListener('DOMContentLoaded', getRecs);
+  document.addEventListener('DOMContentLoaded', load);
 } else {  // `DOMContentLoaded` has already fired
+  l();
```

Comment:
I'm not too sure how callback functions work, but i think you could call `load()` here instead, if that was your reason for using the `l` variable above.